app/services/oneoffs/npq/data_cleanup/fix_applications_in_ecf_not_in_npq.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module Oneoffs::NPQ::DataCleanup
+  class FixApplicationsInECFNotInNPQ
+    attr_reader :npq_application_ids
+
+    def initialize(npq_application_ids:)
+      @npq_application_ids = npq_application_ids
+    end
+
+    def run!(dry_run: true)
+      result = {}
+
+      ActiveRecord::Base.transaction do
+        result = npq_application_ids.each_with_object({}) do |npq_application_id, hash|
+          npq_application = NPQApplication.find_by(id: npq_application_id)
+          next unless npq_application
+
+          # Other applications for the same user/course/lead_provider/cohort
+          similar_applications = similar_applications(npq_application:)
+
+          # Where there is only one accepted similar application
+          # We keep the accepted one and delete the one which does not exist in NPQ.
+          accepted_applications = similar_applications.select(&:accepted?)
+          if accepted_applications.size == 1 && !npq_application.accepted?
+            npq_application.destroy!
```

Comment:
If we have duplicates in both NPQ and ECF, how can we ensure we keep those in both as we agreed?